src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -435,14 +367,74 @@ namespace AppInstaller::CLI::Workflow
                 context.Reporter.Warn() << Resource::String::OverwritingExistingFileAtMessage << ' ' << symlinkFullPath.u8string() << std::endl;
             }
 
-            std::filesystem::create_symlink(targetFullPath, symlinkFullPath);
-            AICLI_LOG(CLI, Info, << "Symlink created at: " << symlinkFullPath);
-            uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, symlinkFullPath.wstring());
+            if (Filesystem::CreateSymlink(targetFullPath, symlinkFullPath))
+            {
+                AICLI_LOG(CLI, Info, << "Symlink created at: " << symlinkFullPath);
+                uninstallEntry.SetValue(PortableValueName::PortableSymlinkFullPath, symlinkFullPath.wstring());
```

Comment:
The general flow for these should be:
1. Write artifact stating that you added a file
2. Attempt to add the file
3. <Time passes, user uninstalls...>
4. Attempt to delete the file due to presence of artifact (if it points to something valid)
5. Remove artifact

If you wait to write the artifact until after you add the file, you could abandon it due to failing to write the artifact or a crash or a system power loss.  If you write the artifact and it points to nothing due to one of the error states, then step 4 is a check only and nothing is left behind.

I agree that it makes things more complicated, but the goal is to create a robust system, and that robustness is less efficient in both code and runtime.